### ISSUE TITLE
Rich data analysis

### ISSUE THREAD

**Issue by lambdafu** [das-labor/panopticon#95]
There needs to be a framework and some examples how to analyse data ranges.  There is a continuum from basic display options (raw hex, decimal, binary, number of bytes per line) to specialised applications such as embedded icons and images, audio (actually any mime type), and ending in application-specific formats (C64 games with embedded resources in ad-hoc formats).

But for a start, there needs to be something.  See also #91 #92 #93 

The metadata should also end up in a human-readable form in exported output formats, cf. #94

**Comment by flanfly** [comment]
The C++ version had an incomplete API for structure definions: https://github.com/das-labor/panopticon/blob/f94458940270cd40a12be597ff86a369950dd755/lib/include/panopticon/format.hh

**Comment by lambdafu** [comment]
Yeah, some basic hard-coded support like that and maybe JS-snippets attached to ranges in the hexview for user scripting?  Even a very basic JS interface could be very useful, although of course the core would not really be able to pick up on the output of JS snippets unless there is a well-defined back channel.  Mh.

**Comment by flanfly** [comment]
I opened a new issue for embedding a scripting language into Panopticon: #103. I don't expect that this will be implemented any time soon.

For now it would be possible to port the old format code and add an C parser. The user could define his structure definitions with just plain C `struct`, `union` and `typedef`s. I believe this would cover 80% of all use cases.
